libs/datasets/nytimes_anomalies.py

```diff
@@ -0,0 +1,179 @@
+
+    # Convert "type" column into "variable" column using new_cases / new_deaths as the variable.
+    assert df[NYTimesFields.TYPE].isin(["cases", "deaths"]).all()
+    df[PdFields.VARIABLE] = df[NYTimesFields.TYPE].map(
+        {"cases": CommonFields.NEW_CASES, "deaths": CommonFields.NEW_DEATHS}
+    )
+
+    # Add demographic bucket (all) to make it more compatible with our dataset structure.
+    df[PdFields.DEMOGRAPHIC_BUCKET] = "all"
+
+    return df
+
+
+# TODO(mikelehen): This should probably live somewhere more central, but I'm not sure where.
+def _get_county_fips_codes_for_state(state_fips_code: str) -> List[str]:
+    """Helper to get county FIPS codes for all counties in a given state."""
+    geo_data = dataset_utils.get_geo_data()
+    state = geo_data.set_index("fips").at[state_fips_code, "state"]
+    counties_df = geo_data.loc[
+        (geo_data["state"] == state) & (geo_data["aggregate_level"] == "county")
+    ]
+    counties_fips = counties_df["fips"].to_list()
+    return counties_fips
+
+
+def _denormalize_nyt_anomalies(df: pd.DataFrame) -> pd.DataFrame:
+    """
+    The NYT anomaly data is normalized such that each row can represent an
+    anomaly for multiple dates, locations, and metrics. We want to denormalize
+    it so that each row represents a single date+location+metric anomaly.
+    """
+
+    # Look for rows with an end_date and create separate rows for each date in the [date, end_date] range.
+    df = df.assign(
+        date=df.apply(
+            lambda row: pd.date_range(
+                row[NYTimesFields.DATE],
+                row[NYTimesFields.DATE]
+                if pd.isna(row[NYTimesFields.END_DATE])
+                else row[NYTimesFields.END_DATE],
+                freq="d",
+            ),
+            axis=1,
+        ),
+        end_date=pd.NaT,
+    ).explode(NYTimesFields.DATE)
+
+    # Look for state rows with omit_from_rolling_average_on_subgeographies and add new rows for each county within the state
+    omit_subgeographies_mask = (
+        df[NYTimesFields.OMIT_FROM_ROLLING_AVERAGE_ON_SUBGEOGRAPHIES] == "yes"
+    ) & (df[CommonFields.FIPS].str.len() == 2)
+    df_subgeography_rows = df.loc[omit_subgeographies_mask].copy()
+    df_subgeography_rows = df_subgeography_rows.assign(
+        fips=df_subgeography_rows.apply(
+            lambda row: _get_county_fips_codes_for_state(row[CommonFields.FIPS]), axis=1
+        ),
+        omit_from_rolling_average="yes",
+    ).explode(CommonFields.FIPS)
+    df = df.append(df_subgeography_rows)
+
+    # Look for rows with type=='both' and replace with two rows ('cases' and 'deaths')
+    df = df.assign(
+        type=df.apply(
+            lambda row: ["cases", "deaths"]
+            if row[NYTimesFields.TYPE] == "both"
+            else row[NYTimesFields.TYPE],
+            axis=1,
+        )
+    ).explode(NYTimesFields.TYPE)
+
+    return df.reset_index()
+
+
+def filter_by_nyt_anomalies(dataset) -> MultiRegionDataset:
+    """
```

Comment:
This is a nit, but pointing out this docstring pattern which we've (or at least i have) tried to be pretty consistent with:

```
"""<summary line>.

<more description if you want>.

"""
```

In general the Google style guide specifies that the first line should be a heading line and stick to just one line, then more description can come after it.

Reply:
Sure, done.